packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -809,58 +809,70 @@ class _LargeTitleNavigationBarSliverDelegate
-                      top: false,
-                      bottom: false,
-                      child: AnimatedOpacity(
-                        opacity: showLargeTitle ? 1.0 : 0.0,
-                        duration: _kNavBarTitleFadeDuration,
-                        child: Semantics(
-                          header: true,
-                          child: DefaultTextStyle(
-                            style: CupertinoTheme.of(context).textTheme.navLargeTitleTextStyle,
-                            maxLines: 1,
-                            overflow: TextOverflow.ellipsis,
-                            child: components.largeTitle!,
+      child: LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
+          if (math.min(1.15, 1 + (constraints.maxHeight - maxExtent)/maxExtent*0.12) == 1.15)
+            print(constraints.maxHeight);
+          return DefaultTextStyle(
+            style: CupertinoTheme.of(context).textTheme.textStyle,
+            child: Stack(
+              fit: StackFit.expand,
+              children: <Widget>[
+                Positioned(
+                  top: persistentHeight,
+                  left: 0.0,
+                  right: 0.0,
+                  bottom: 0.0,
+                  child: ClipRect(
+                    // The large title starts at the persistent bar.
+                    // It's aligned with the bottom of the sliver and expands clipped
+                    // and behind the persistent bar.
+                    child: OverflowBox(
+                      minHeight: 0.0,
+                      maxHeight: double.infinity,
+                      alignment: AlignmentDirectional.bottomStart,
+                      child: Padding(
+                        padding: const EdgeInsetsDirectional.only(
+                          start: _kNavBarEdgePadding,
+                          bottom: 8.0, // Bottom has a different padding.
+                        ),
+                        child: SafeArea(
+                          top: false,
+                          bottom: false,
+                          child: AnimatedOpacity(
+                            opacity: showLargeTitle ? 1.0 : 0.0,
+                            duration: _kNavBarTitleFadeDuration,
+                            child: Semantics(
+                              header: true,
+                              child: DefaultTextStyle(
+                                style: CupertinoTheme.of(context).textTheme.navLargeTitleTextStyle,
+                                maxLines: 1,
+                                overflow: TextOverflow.ellipsis,
+                                child: Transform.scale(
+                                  // This scale is estimated from the settings app in iOS 14.
+                                  // It has a maximum magnification of about 15%.
+                                  scale: math.min(1.15, 1 + (constraints.maxHeight - maxExtent)/maxExtent*0.12),
```

Comment:
Let the comment describe the breakdown a bit. What's the formula trying to do. 

Nit for style: spaces around operators.